packages/nimble-components/src/anchor-menu-item/index.ts

```diff
@@ -0,0 +1,110 @@
+import { attr, observable } from '@microsoft/fast-element';
+import {
+    DesignSystem,
+    AnchorOptions,
+    MenuItem as FoundationMenuItem
+} from '@microsoft/fast-foundation';
+import { keyEnter } from '@microsoft/fast-web-utilities';
+import { AnchorBase } from '../anchor-base';
+import { styles } from './styles';
+import { template } from './template';
+
+declare global {
+    interface HTMLElementTagNameMap {
+        'nimble-anchor-menu-item': AnchorMenuItem;
+    }
+}
+
+/**
+ * Types of anchor menu item column count.
+ * This is how many grid columns precede the menu item text.
+ * 1 corresponds to having space for an icon, 0 corresponds to the text being flush left.
+ * Users should not need to use this type. The menu sets startColumnCount on its items.
+ * @public
+ */
+export type AnchorMenuItemColumnCount = 0 | 1;
+
+/**
+ * A nimble-styled anchor menu-item
+ */
+export class AnchorMenuItem extends AnchorBase {
+    @attr({ mode: 'boolean' })
+    public disabled = false;
+
+    @observable
+    public anchor!: HTMLAnchorElement;
+
+    @observable
+    public startColumnCount: AnchorMenuItemColumnCount = 0;
```

Comment:
Both of these should be least privledge / `@internal` since they are not part of the user-facing public api